### PR TITLE
Perform hosts input validation when creating/updating network profiles

### LINKED ISSUE #77: Perform hosts input validation when creating/updating network profiles.
## Specify type:
- Enhancement
___

## Description:
Should validate the input for various acceptable types.
- ip address
- CIDR range string
- Ansible range string
- hostname

(similar validation that was done in `rho`).

___
Associated with https://github.com/quipucords/quipucords/issues/23

### PR DESCRIPTION
**Pull request by kholdaway**
Closes #77

### CODE REVIEW

File: quipucords/api/tests_networkprofile.py [hunk 1]
```diff
@@ -104,6 +104,43 @@ def test_create_empty_host(self):
...
+                       '10.10.[1:20].[1:25]',
+                       'localhost',
+                       'mycentos.com',
+                       'my-rhel[a:d].company.com',
+                       'my-rhel[120:400].company.com'],
+             'ssh_port': '22',
+             'credentials': [self.cred_for_upload]})
+
+    def test_create_invalid_hosts(self):
+        """An empty string is not a host identifier."""
```
Comment (chambridge) [inline quipucords/api/tests_networkprofile.py]: Doesn't seem to be what the test is about.

File: quipucords/api/tests_networkprofile.py [hunk 2]
```diff
@@ -104,6 +104,43 @@ def test_create_empty_host(self):
...
+                       '192.3.4.455',
+                       '192.3.4.455/16',
+                       '10.10.[181.9',
+                       '10.10.128.[a:25]',
+                       '10.10.[1-20].25',
+                       '1.1.1.1/33',
+                       'my_rhel[a:d].company.com',
+                       'my-rhel[a:400].company.com'],
+             'ssh_port': '22',
+             'credentials': [self.cred_for_upload]})
```
Comment (chambridge) [inline quipucords/api/tests_networkprofile.py]: Maybe we should have a (smaller) test that checks the error message is as expected?

File: quipucords/api/networkprofile_serializer.py [hunk 3]
```diff
@@ -131,14 +134,181 @@ def validate_name(name):
...
+                if result is None:
+                    result = ValidationError(
+                        '{} is invalid host'.format(host_range))
+
+            if isinstance(result, ValidationError):
+                host_errors.append(result)
+            elif result is not None:
+                normalized_hosts.append(result)
+            else:
+                logging.error('%s did not match a pattern or produce error',
```
Comment (chambridge) [inline quipucords/api/networkprofile_serializer.py]: I feel like we should be capturing some kind of unknown format or cannot process validation error here.

File: quipucords/api/tests_networkprofile.py (hunk 1 above)
Comment (kholdaway) [reply quipucords/api/tests_networkprofile.py]: Modified test to ensure each of the hosts are invalid

**Pull request merged**